compiler/z/codegen/OMRCodeGenerator.cpp

```diff
@@ -715,14 +715,42 @@ OMR::Z::CodeGenerator::CodeGenerator()
    self()->setSupportsBCDToDFPReduction();
    self()->setSupportsIntDFPConversions();
 
+   /* Using traps on z/OS for NullPointerException and ArrayIndexOutOfBound checks instead of the
+    * old way of using explicit compare and branching off to a helper is causing several issues on z/OS:
+    *
+    * 1. When a trap fires because an exception needs to be thrown, the OS signal has to go through z/OS RTM
+    * (Recovery Termination Management) and in doing so it gets serialized and has to acquire this CML lock.
+    *  When many concurrent exceptions are thrown, this lock gets contention.
+    *
+    * 2. on z/OS Management Facility, disabling traps gave performance boost because of fewer exceptions from the JCL.
+    * The path length on z/OS for a trap is longer than on other operating systems so the trade-off and
+    * penalty for using traps on z/OS is significantly worse than other platforms. This gives the potential
+    * for significant performance loss caused by traps.
+    *
+    * 3. Depending on sysadmin settings, every time a trap fires, the system may get a "system dump"
+    * message which shows a 0C7 abend and with lots of exceptions being thrown this clutters up the syslog.
+    *
+    * 4. Certain products can get in the way of the JIT signal handler so the JIT doesn't
+    * receive the 0C7 signal causing the JVM process to get killed
+    *
+    * Therefore the recommendation is to disable traps on z/OS.
+    */
+   if (TR::Compiler->target.isZOS() && !comp->getOptions()->getOption(TR_EnableTraps))
+      {
+      comp->setOption(TR_DisableTraps);
+      }
+
    if (_processorInfo.supportsArch(TR_S390ProcessorInfo::TR_z10))
       {
       self()->setSupportsTranslateAndTestCharString();
 
       self()->setSupportsBCDToDFPReduction();
       self()->setSupportsIntDFPConversions();
 
-      if (!comp->getOptions()->getOption(TR_NoResumableTrapHandler) && TR::Compiler->vm.hasResumableTrapHandler(comp))
+      // change from checking the noResumableTrapHandler to the new disableTrap option to decouple disabling
+      // the handler which is also used by DAA with disabling trap instructions. However, keep the
```

Comment:
This comment needs an update. There is no concept of "DAA" or "service" in OMR. I would in fact just remove the comment all together as the above explanation is more than enough.